lib/sles4sap.pm

```diff
@@ -977,12 +989,482 @@ sub startup_type {
+
+=cut
+
+sub is_instance_type_supported {
+    my ($self, $instance_type) = @_;
+    my @supported_values = qw(ASCS HDB ERS PAS AAS);
+    croak("Argument '\$instance_type' undefined")
+      unless $instance_type and grep(/^$instance_type$/, @supported_values) > 0;
+    return $instance_type;
+}
+
+=head2 share_hosts_entry
+
+ $self->share_hosts_entry(virtual_hostname=>$virtual_hostname,
+    virtual_ip=>$virtual_ip,
+    shared_directory_root=>shared_directory_root);
+
+ Creates file with virtual IP and hostname entry for /etc/hosts file on mounted shared device (Default: /sapmnt).
+ This is to help creating /etc/hosts file which would include entries for all nodes.
+ File name: <INSTANCE_TYPE>
+ File content: <virtual IP> <virtual_hostname>
+
+B<virtual_hostname> Virtual hostname (alias) which is tied to instance and will be moved with HA IP addr resource
+B<virtual_ip> Virtual IP addr tied to an instance and HA resource
+B<shared_directory_root> Shared directory available for all instances. Separate directory 'hosts' will be created
+
+=cut
+
+sub share_hosts_entry {
+    my ($self, %args) = @_;
+    my $virtual_hostname = $args{virtual_hostname} // get_var('INSTANCE_ALIAS', '$(hostname)');
+    my $instance_type = get_required_var('INSTANCE_TYPE');
+    my $virtual_ip = $args{virtual_ip} // (split('/', get_required_var('INSTANCE_IP_CIDR')))[0];
+    my $shared_directory_root = $args{shared_directory_root} // '/sapmnt';    # sapmnt is usually shared between all hosts
+    set_var('HOSTS_SHARED_DIRECTORY', "$shared_directory_root/hosts");
+
+    croak "Directory '$shared_directory_root' does not appear to be a mount point" if
+      assert_script_run("mount | grep $shared_directory_root");
+
+    assert_script_run("mkdir -p $shared_directory_root/hosts");
+    assert_script_run("echo '$virtual_ip $virtual_hostname' >> $shared_directory_root/hosts/$instance_type");
+}
+
+=head2 add_hosts_file_entries
+
+ $self->add_hosts_file_entries();
+
+Reads files in 'HOSTS_SHARED_DIRECTORY' and adds entries into /etc/hosts file.
+
+=cut
+
+sub add_hosts_file_entries {
+    my $source_directory = get_var('HOSTS_SHARED_DIRECTORY', '/sapmnt/hosts');    # sapmnt is usually shared between all hosts
+    my @files = split(' ', script_output("ls $source_directory"));
+    for my $file (@files) {
+        assert_script_run("cat $source_directory/$file >> /etc/hosts");
+    }
+    assert_script_run('cat /etc/hosts');
+}
+
+=head2 get_sidadm
+
+ $self->get_sidadm([must_exist=>$must_exist]);
+
+Returns sidadm username created from SAP sid - parameter INSTANCE_SID.
+check_if_exists - if set to true, test will fail if user does not exist. tests inconsistency between OpenQA parameter and real username
+
+B<must_exist> Checks if sidadm exists, croaks on failure. Default 'false'
+
+=cut
+
+sub get_sidadm {
+    my ($self, %args) = @_;
+    my $must_exist = $args{'must_exist'} // 0;
+    my $sidadm = lc get_required_var('INSTANCE_SID') . 'adm';
```

Comment:
Yeah I know. I was trying to avoid having many dependencies between modules/functions to make changes in the code easier. 
set_sap_info is doing multiple not necessarily related things and for example requires instance no or variable 'INSTANCE_TYPE' which are not necessary if you need only sidadm user... let's say for ssh to another host or calculating another variable from username.  I just wanted a function that does one thing. 
Is it a big problem if I wanna keep it ?